app/src/main/java/com/nutomic/syncthingandroid/service/EventProcessor.java

```diff
@@ -242,10 +209,80 @@ public void stop() {
         }
     }
 
-    private void notify(String text, PendingIntent pi) {
-        // HACK: Use a random, deterministic ID between 1000 and 2000 to avoid duplicate
-        //       notifications.
-        int notificationId = 1000 + text.hashCode() % 1000;
-        mNotificationHandler.showEventNotification(text, pi, notificationId);
+    private void onDeviceRejected(String deviceId, String deviceName) {
+        if (deviceId == null) {
+            return;
+        }
+        Log.d(TAG, "Unknown device " + deviceName + "(" + deviceId + ") wants to connect");
+
+        String title = mContext.getString(R.string.device_rejected,
+                deviceName.isEmpty() ? deviceId.substring(0, 7) : deviceName);
+        int notificationId = mNotificationHandler.getNotificationIdFromText(title);
+
+        // Prepare "accept" action.
+        Intent intentAccept = new Intent(mContext, DeviceActivity.class)
+                .putExtra(DeviceActivity.EXTRA_NOTIFICATION_ID, notificationId)
+                .putExtra(DeviceActivity.EXTRA_IS_CREATE, true)
+                .putExtra(DeviceActivity.EXTRA_DEVICE_ID, deviceId)
+                .putExtra(DeviceActivity.EXTRA_DEVICE_NAME, deviceName);
+        // Use a deterministic ID to make multiple PendingIntents distinguishable.
+        int requestCode = deviceId.hashCode();
+        PendingIntent piAccept = PendingIntent.getActivity(mContext, requestCode,
+            intentAccept, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        // Prepare "ignore" action.
+        Intent intentIgnore = new Intent(mContext, SyncthingService.class)
+                .putExtra(SyncthingService.EXTRA_NOTIFICATION_ID, notificationId)
+                .putExtra(SyncthingService.EXTRA_DEVICE_ID, deviceId);
+        intentIgnore.setAction(SyncthingService.ACTION_IGNORE_DEVICE);
+        PendingIntent piIgnore = PendingIntent.getService(mContext, 0,
+            intentIgnore, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        // Show notification.
+        mNotificationHandler.showConsentNotification(notificationId, title, piAccept, piIgnore);  // ToDo
+    }
+
+    private void onFolderRejected(String deviceId, String folderId,
+                                    String folderLabel) {
+        if (deviceId == null || folderId == null) {
+            return;
+        }
+        Log.d(TAG, "Device " + deviceId + " wants to share folder " +
+            folderLabel + " (" + folderId + ")");
+
+        // Find the deviceName corresponding to the deviceId
+        String deviceName = null;
+        for (Device d : mApi.getDevices(false)) {
+            if (d.deviceID.equals(deviceId))
+                deviceName = d.getDisplayName();
```

Comment:
break?

Reply:
Agreed. I've overseen this taking over the already existing code.